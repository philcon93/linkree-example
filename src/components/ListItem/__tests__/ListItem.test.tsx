import { fireEvent, render } from '@testing-library/react';
import { ListItem } from '../';

describe('ListItem', () => {
    test('Required props work correctly', () => {
        const mockOnClick = jest.fn();

        const { container, getByText, getByTestId } = render(
            <ListItem title={'title text'} onClick={mockOnClick}/>
        );

        expect(container).toMatchSnapshot();
        expect(getByText('title text')).toBeInTheDocument();

        fireEvent.click(getByTestId('ListItem'));
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});
