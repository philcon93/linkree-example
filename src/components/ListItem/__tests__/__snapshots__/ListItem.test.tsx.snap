// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem Required props work correctly 1`] = `
<div>
  <div
    class="css-70qvj9"
  >
    <div
      class="css-th0ams"
      data-testid="ListItem"
    >
      <div
        class="css-0"
      >
        <p
          class="chakra-text css-504jus"
        >
          title text
        </p>
      </div>
      <div
        class="css-17xejub"
      />
      <img
        class="chakra-image__placeholder css-jbc4oa"
      />
    </div>
  </div>
</div>
`;
